src/Agent.Worker/Build/FileContainerServer.cs

```diff
@@ -204,6 +204,14 @@ private async Task<UploadResult> ParallelUploadAsync(IAsyncCommandContext contex
             _uploadFinished.TrySetResult(0);
             await uploadMonitor;
 
+            // report telemetry
+            if (uploadToBlob)
+            {
+                var planId = Guid.Parse(context.GetVariableValueOrDefault(WellKnownDistributedTaskVariables.PlanId) ?? Guid.Empty.ToString());
```

Comment:
Could Guid.Parse throw here?  Should we be doing TryParse?

Reply:
Done